python/cudf/cudf/core/udf/pipeline.py

```diff
@@ -16,7 +18,48 @@
 precompiled: cachetools.LRUCache = cachetools.LRUCache(maxsize=32)
 
 
-def get_frame_row_type(fr):
+class _FrameJitMetadata(object):
+    def __init__(self, frame):
+        """
+        convenience class that contains the metadata from the
+        frame the UDF is targeting that is needed for jitting
+        the eventual kernel.
+        """
+        self.all_dtypes = {}
+        self.supported_dtypes = {}
+        self.frame = frame
+
+        for colname, col in self.frame._data.items():
+            if _is_jit_supported_type(col.dtype):
+                self.supported_dtypes[colname] = col.dtype
+                np_type = col.dtype
+            else:
+                np_type = np.dtype("O")
+            self.all_dtypes[colname] = np_type
+
+        self.supported_cols = {
+            name: self.frame._data[name]
+            for name in self.supported_dtypes.keys()
+        }
+
+
+def generate_cache_key(frame_meta: _FrameJitMetadata, func: Callable):
+    """
+    Create a cache key that uniquely identifies a compilation. A new
+    compilation is needed any time any of the following things change:
```

Comment:
For easing future pydocstyle compliance:
```suggestion
    """Create a cache key that uniquely identifies a compilation.

    A new compilation is needed any time any of the following things change:
```